keccak/src/lib.rs

```diff
@@ -145,11 +150,17 @@ impl_keccak!(f200, u8);
 impl_keccak!(f400, u16);
 impl_keccak!(f800, u32);
 
-#[cfg(not(all(target_arch = "aarch64", target_feature = "sha3")))]
+#[cfg(not(all(target_arch = "aarch64", feature = "asm")))]
 impl_keccak!(f1600, u64);
 
-#[cfg(all(target_arch = "aarch64", target_feature = "sha3"))]
-pub use aarch64_sha3::keccak_f1600 as f1600;
+#[cfg(all(target_arch = "aarch64", feature = "asm"))]
+pub fn f1600(state: &mut [u64; PLEN]) {
+    if armv8_sha3_intrinsics::get() {
```

Comment:
@newpavlov there's not really a way to make this interface work with an init token. Does it seem ok to you?

Reply:
Yeah, it's not ideal, but should be fine. One potential alternative is to expose an unsafe gated `f1600_aarch64_asm` function and do the switch inside `sha3`.

Reply:
Sure, that works. I can update the PR.

Reply:
I am not saying a different function is a better approach. Just floating an idea.

Reply:
It seems nice to have to me, especially for the `sha3` use case.

It's `unsafe` and `#[target_feature(enable = "sha3")]`-gated, which should prevent casual misuse.